git-gui--askpass

```diff
@@ -26,10 +26,25 @@ pack .m -side top -fill x -padx 20 -pady 20 -expand 1
 entry .e -textvariable answer -width 50
 pack .e -side top -fill x -padx 10 -pady 10
 
+proc on_hide_input_changed {args} {
+	global hide_input
+	if {$hide_input} {
+		.e configure -show "*"
+	} else {
+		.e configure -show ""
+	}
+}
+trace add variable hide_input write "on_hide_input_changed"
+
+set hide_input 0
```

Comment:
I think we should hide the input by default and let users show it if they want, and skip the whole "Password" or "passphrase" check. I'm not convinced special casing "Password" and "passphrase" is a good idea. There might be other terms used in the prompt, or even other languages. We can't catch all the cases.

This patch looks good to me otherwise. I do have one nitpick. IMO we should label the button "Show Input" instead of "Hide Input".